### DISCUSSION TITLE
#Nullable and Records

### DISCUSSION THREAD

**Discussion by TonyValenti** [dotnet/csharplang#3999] [Q&A] [answered]
Sample Code:
```
    public record ActivityGlance {
        public Activity Activity { get; init; }
        public Glance Glance { get; init; }
    }

    public record Activity { }
    public record Glance { }
```

What do I get the warnings "Non-nullable property ___ must contain a non-null value...."?

I thought that records would ensure that those properties were set prior to use.

**Comment by HaloFour** [comment, marked as answer]
See https://github.com/dotnet/csharplang/issues/3630

The `init` keyword doesn't require that the property must be initialized, only that the property can only be set during initialization.

**Reply by TonyValenti** [reply]
Thanks!
